### PR TITLE
Add hist

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by henryiii** [comment]
Do we need to manually be passing `--no-build-isolation` for PEP 518 packages, or does conda-forge already set `PIP_NO_BUILD_ISOLATION`?

**Comment by henryiii** [comment]
Also, please add me as a maintainer. Thanks :)

**Comment by LovelyBuggies** [comment]
> Also, please add me as a maintainer. Thanks :)

And me, too :)

**Comment by chrisburr** [comment]
@conda-forge/staged-recipes, ready for review

> Do we need to manually be passing `--no-build-isolation` for PEP 518 packages, or does conda-forge already set `PIP_NO_BUILD_ISOLATION`?

These days conda-build takes care of everything.

File: recipes/hist/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - boost-histogram ~=0.11.0
+    - histoprint >=1.4
+    - numpy >=1.13.3
+    - python
+    - typing-extensions  # [py<38]
```
Comment (synapticarbors) [inline recipes/hist/meta.yaml:29]: I'm wondering if we can remove the py version restriction if this falls back to a reasonable choice of builtins when using py3.8. The package recipe lists this as `noarch: python`

File: recipes/hist/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/hist-{{ version }}.tar.gz
+  sha256: 48edff8f498ea7c5c6d8057c6f001a2ce394acd0923b921c56b76fb732e11df3
+
+build:
+  number: 0
+  skip: true   # [py2k]
```
Comment (synapticarbors) [inline recipes/hist/meta.yaml:15]: conda-forge doesn't support py2 by default, so I think we can get rid of this and make this `noarch: python` (but see comment on `typing-extensions`) and just change the `python` to `python >=3.6` under the `host` and `run` sections of `requirements`.

File: recipes/hist/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  entry_points:
+    - hist=hist.classichist:main
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - boost-histogram ~=0.11.0
```
Comment (synapticarbors) [inline recipes/hist/meta.yaml:25]: I'm not sure if conda handles the `~=` properly, or if it does it's undocumented and thus not reliable. I think this is the same as `>=0.11.0,<0.12.0`.

File: recipes/hist/meta.yaml (hunk 1 above)
Comment (henryiii) [reply recipes/hist/meta.yaml:29]: It does.

File: recipes/hist/meta.yaml (hunk 3 above)
Comment (chrisburr) [reply recipes/hist/meta.yaml:25]: It is [officially supported](https://github.com/conda/conda/pull/7783) and this receipe was generated by grayskull so if it's a problem grayskull should be modified accordingly.

**Pull request merged**